src/scripts/Battle.ts

```diff
@@ -108,7 +108,7 @@ class Battle {
      */
     public static generateNewEnemy() {
         this.counter = 0;
-        this.enemyPokemon(PokemonFactory.generateWildPokemon(player.route(), player.region));
+        this.enemyPokemon(PokemonFactory.generateWildPokemon(player.route(), player.region, SubRegions.getSubRegionById(player.region, player.subregion)));
```

Comment:
Should we be storing the current SubRegion object somewhere so we won't need to call the `SubRegions.getSubRegionById` every time we generate a Pokémon?
Similar to how we store `player.town`